lib/kubernetes-deploy/restart_task.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+require 'kubernetes-deploy/kubeclient_builder'
+
+module KubernetesDeploy
+  class RestartTask
+    include KubernetesDeploy::KubeclientBuilder
+
+    def initialize(context:, namespace:, logger:)
+      @context = context
+      @namespace = namespace
+      @logger = logger
+      @v1beta1_kubeclient = build_v1beta1_kubeclient(context)
+    end
+
+    def perform(deployments)
+      if deployments.empty?
+        raise ArgumentError, "#perform takes at least one deployment to restart"
+      end
+
+      fetch_deployments(deployments).each do |record|
+        response = patch_deployment_with_restart(record)
+        if response.code == 200
+          @logger.info "Restarted #{record.metadata.name.inspect}"
+        else
+          @logger.fatal "Failed to restart #{record.metadata.name} (API returned non-200 response)"
+          @logger.fatal "Response:"
+          @logger.fatal response.body
+        end
+      end
+    end
+
+    private
+
+    def build_v1beta1_kubeclient(context)
+      build_kubeclient(
+        api_version: "v1beta1",
+        context: context,
+        endpoint_path: "/apis/extensions/"
+      )
+    end
+
+    def patch_deployment_with_restart(record)
+      @v1beta1_kubeclient.patch_deployment(
+        record.metadata.name,
+        build_patch_payload(record),
+        @namespace
+      )
+    end
+
+    def fetch_deployments(list)
+      list.map do |name|
+        records = @v1beta1_kubeclient.get_deployments(label_selector: "name=#{name}", namespace: @namespace)
+        raise "too many records: #{records.size}" if records.size > 1
```

Comment:
That's actually an exceptional case (two deployments with name `web` in the same namespace). I'm not sure that can ever happen in production, but I'd prefer to fail rather than proceed.